torch/nn/functional.py

```diff
@@ -2783,6 +2783,36 @@ def pairwise_distance(x1, x2, p=2., eps=1e-6, keepdim=False):
     >>> print(output)
 """)
 
+@weak_script
+def squareform(input):
+    # type: (Tensor) -> Tensor
+    r"""Convert a vector-form distance vector to a square-form distance matrix, and vice-versa.
+
+    Args:
+        input (Tensor): `1d` tensor or `2d` tensor
+
+    Examples::
+    >>> vec = torch.arange(1, 11)
+    >>> vec
+    tensor([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10])
+    >>> sqm = F.squareform(vec)
+    >>> sqm
+    array([[ 0,  1,  2,  3,  4],
+           [ 1,  0,  5,  6,  7],
+           [ 2,  5,  0,  8,  9],
+           [ 3,  6,  8,  0, 10],
+           [ 4,  7,  9, 10,  0]])
+    >>> F.squareform(sqm)
+    tensor([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10])
+
+    """
+    assert input.dim() == 1 or input.dim() == 2, 'dimension of the input need be 1 or 2'
```

Comment:
squareform should be a function at the C++ (native_function) level.  The downside of doing it this way is that the C++ API does not match the python API.